media-api/app/lib/elasticsearch/QueryBuilder.scala

```diff
@@ -116,7 +111,15 @@ class QueryBuilder(matchFields: Seq[String], overQuotaAgencies: () => List[Agenc
 
         }.toList
 
-      nestedQueries.foldLeft(query) { case (q, nestedQ) => q.withMust(nestedQ) }
+      val queryWithNormal = normal.foldLeft(boolQuery()) {
+        case (query, Negation(cond)) => query.withNot(makeQueryBit(cond))
+        case (query, cond@Match(_, _)) => query.withMust(makeQueryBit(cond))
+        case (query, _) => query
+      }
+
+      val queryWithNestedAndNormal = listOfNestedToQueries(nested).foldLeft(queryWithNormal) { case (q, nestedQ) => q.withMust(nestedQ) }
+
+      listOfNestedToQueries(negationNested).foldLeft(queryWithNestedAndNormal) { case (q, negNestedQ) => q.withNot(negNestedQ) }
```

Comment:
I can just about make out what's happening here but it could do with some more clarity?